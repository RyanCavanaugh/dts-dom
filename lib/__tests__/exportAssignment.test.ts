import {ContextFlags, create, emit} from "../index"

describe("an export assignment", () => {
    describe("with isExportEquals set to true", () => {
        it("matches the snapshot", () => {
            const exportAssignment = create.exportAssignment('test', true);
            expect(emit(exportAssignment)).toMatchSnapshot();
        });
    });

    describe("with isExportEquals set to false", () => {
        it("matches the snapshot", () => {
            const exportAssignment = create.exportAssignment('test', false);
            expect(emit(exportAssignment)).toMatchSnapshot();
        });
    });

    describe("with isExportEquals omitted", () => {
        it("matches the snapshot", () => {
            const exportAssignment = create.exportAssignment('test');
            expect(emit(exportAssignment)).toMatchSnapshot();
        });
    });

    describe("as module member", () => {
        it("matches the snapshot", () => {
            const module = create.module('my-module');
            const constDeclaration = create.const('test', 'string');
            const exportAssignment = create.exportAssignment('test');
            module.members.push(constDeclaration, exportAssignment);
            expect(emit(module)).toMatchSnapshot();
        });
    });
});
