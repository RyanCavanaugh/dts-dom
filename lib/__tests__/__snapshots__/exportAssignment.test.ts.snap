// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`an export assignment as module member matches the snapshot 1`] = `
declare module 'my-module' {
    declare const test: string;

    export default test;

}


`;

exports[`an export assignment with isExportEquals omitted matches the snapshot 1`] = `
export default test;


`;

exports[`an export assignment with isExportEquals set to false matches the snapshot 1`] = `
export default test;


`;

exports[`an export assignment with isExportEquals set to true matches the snapshot 1`] = `
export = test;


`;
